lib/active_fedora/simple_datastream.rb

```diff
@@ -75,25 +75,20 @@ def update_indexed_attributes(params={}, opts={})
       end
       super(new_params, opts)
     end
-    
 
     def self.xml_template
-       Nokogiri::XML::Document.parse("<fields/>")
+      Nokogiri::XML::Document.parse("<fields/>")
     end
 
     def to_solr(solr_doc = Hash.new) # :nodoc:
       @fields.each do |field_key, field_info|
         next if field_key == :location ## FIXME HYDRA-825
-        things = send(field_key)
-        if things 
-          field_symbol = ActiveFedora::SolrService.solr_name(field_key, type: field_info[:type])
-          things.val.each do |val|    
-            ::Solrizer::Extractor.insert_solr_field_value(solr_doc, field_symbol, val.to_s )         
-          end
+        field_symbol = ActiveFedora::SolrService.solr_name(field_key, type: field_info[:type])
+        om_term_values(field_key).each do |val|
```

Comment:
`#om_term_values` always returns an Array, so don't bother with the conditional or the variable assignment.